src/Kestrel/ServerFactory.cs

```diff
@@ -36,7 +40,8 @@ public IDisposable Start(IServerInformation serverInformation, Func<IFeatureColl
         {
             var disposables = new List<IDisposable>();
             var information = (ServerInformation)serverInformation;
-            var engine = new KestrelEngine(_libraryManager);
+            var loggerFactory = _serviceProvider.GetRequiredService<ILoggerFactory>();
```

Comment:
It works well. But, I agree with you, automatically resolve from DI more efficient way.